src/openzeppelin/utils/presets/UniversalDeployer.cairo

```diff
@@ -0,0 +1,56 @@
+%lang starknet
+
+from starkware.starknet.common.syscalls import get_caller_address, deploy
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.hash import hash2
+from starkware.cairo.common.bool import FALSE, TRUE
+
+@event
+func ContractDeployed(
+    address: felt,
+    deployer: felt,
+    classHash: felt,
+    salt: felt
+) {
+}
+
+@external
+func deployContract{
+    syscall_ptr: felt*,
+    pedersen_ptr: HashBuiltin*,
+    range_check_ptr
+}(
+    classHash: felt,
+    salt: felt,
+    unique: felt,
+    calldata_len: felt,
+    calldata: felt*
+) -> (address: felt) {
+    let (deployer) = get_caller_address();
+
+    tempvar prefix;
+    if (unique == TRUE) {
+        prefix = deployer;
+    } else {
+        prefix = 'UniversalDeployerContract';
+    }
+
+    let (_salt) = hash2{hash_ptr=pedersen_ptr}(prefix, salt);
+
+    let (address) = deploy(
+        class_hash=classHash,
+        contract_address_salt=_salt,
+        constructor_calldata_size=calldata_len,
+        constructor_calldata=calldata,
+        deploy_from_zero=FALSE,
+    );
+
+    ContractDeployed.emit(
+        atddress=address,
+        deployer=deployer,
+        classHash=classHash,
+        salt=salt
```

Comment:
this should be `_salt`

Reply:
thanks this was something i wasn't sure about, but why? i leaned towards `salt` since `_salt` is inferable based on either the deployer address (also logged) or the prefix. more so, that property allows for anyone inspecting the event to understand whether `unique` was `TRUE` or `FALSE`.

all in all, i thought `salt` provided more information whereas `salt` was ambiguous/more obscure, but maybe `salt` is more handy? interested to know

Reply:
Both are fine, but I think that if we opt to use `salt`, we should also add the `unique` boolean to the event.

Reply:
What is the intent behind this event? What are we assuming observers of this event are interested in?

IMO, it's more important to have the constructor arguments in the event than `salt`, `_salt`, or `unique`, as I'm assuming the event will be used to find and identify deployed contracts. Observers might want to find all of the deployments of a particular class hash, and know the parameters (constructor arguments) of each, or I might want to search for a particular contract and to find its class hash and parameters.

Regarding the salt, I would favor logging the arguments to the function and the resulting deployed address rather than the intermediate values (`_salt`). It's true that given `_salt` you can infer `unique` but this requires you to know how to compute the deployed address yourself, and I think it's better to allow event observers to be agnostic to that.

Reply:
I needed to validate in SDK dev that precalculated salt was indeed one used by the contract. So I did modify it locally and deploy it to check values.
I think the best solution would be to emit both salt=salt and deploySalt=_salt